projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php

```diff
@@ -189,7 +191,10 @@ public function should_user_see_modal() {
 		}
 
 		// Don't show if post is for subscribers only or has paywall block
-		if ( has_block( 'jetpack/paywall' ) ) {
+		global $post;
+		$access_level              = get_post_meta( $post->ID, META_NAME_FOR_POST_LEVEL_ACCESS_SETTINGS, true );
+		$is_accessible_by_everyone = Token_Subscription_Service::POST_ACCESS_LEVEL_EVERYBODY === $access_level;
+		if ( has_block( 'jetpack/paywall' ) || ! $is_accessible_by_everyone ) {
```

Comment:
I think we could just check `! $is_accessible_by_everyone`

Reply:
@lezama - To confirm (since you've worked with the access code more), does $access_level include or account for if there is a paywall block on the page?

Reply:
we add the post meta whenever someone restricts the access to the post, with or without paywall block